### ISSUE TITLE
Add support for subscripts and superscripts

### ISSUE THREAD

**Issue by sphh** [aardappel/treesheets#121]
I use Treesheets in a technical context and hence I tend to use sub- and superscripts quite a lot. At the moment I am bound to the unicode sub- and superscripts available in the font used. Would it be possible to add sub- and superscript formatting not to whole cells, but to characters/words in cells?

**Comment by aardappel** [comment]
That won't be easy, since currently TreeSheets has no formatting changes inside the cell of any kind, i.e. you can't even make a single world bold, only the entire cell. We'd have to first transition the cell text from plain text to a formatted representation, which is not trivial, since it will affect everything. Frankly that is not likely to happen.

**Comment by sphh** [comment]
Yes, I also noticed that there are no formatting changes inside the cell, but I don't need them as often as sub- and superscripts.

So it is not as easy as replacing `TextCtrl` with `RichTextCtrl`? A pity! So I do understand, that this is sadly not going to happen.

**Comment by aardappel** [comment]
TreeSheets doesn't use any existing gui controls, it is all custom rendering. And it is not even about rendering, it's about the fact that there's a lot of code that assumes the contents of a cell is a simple text string, not some more complex representation.

Probably the easiest way to add formatting, if we'd ever do it, is to keep the text of a cell plain text, but to add a second string with style bits that simply are meant to mirror the ones in the text string. Then functions that don't care about style (such as search) can continue to operate on plain text.

This, as opposed to the more common representation of embedding `bold on` and `bold off` markers in the middle of the text, which is more invasive.

**Comment by sphh** [comment]
Yes, searching should still work without knowing the formatting details of the cell!

**Comment by sphh** [comment]
Hi @aardappel and @tobiolo: Could you please explain your motivation for closing this as not planned for the benefit of us all? Thanks.

**Comment by sphh** [comment]
Well, your https://github.com/aardappel/treesheets/issues/121#issuecomment-512338112 had a positive tone in favour of implementing it:

> Probably the easiest way to add formatting, if we'd ever do it, is to keep the text of a cell plain text, but to add a second string with style bits that simply are meant to mirror the ones in the text string. Then functions that don't care about style (such as search) can continue to operate on plain text.

So I am just surprised, that it now has been closed after 5 year … I admit that I switched from C to Python a long time ago and there the suggested approach of separating display and comparison sounds very reasonable to me (and easy to implement – obviously I do not have any idea at all, what this would involve in this case, because C++ is not a language I speak.

I just find it sad, that such a useful feature will not get included.

(Just a stupid idea: If it were possible to include a cell _inline_ into the contents (=text) of another cell, that would also solve the problem, wouldn't it?)

**Comment by sphh** [comment]
Thank you for the extensive explanation.

Just for the record, I believe that Treesheet would immensely gain usability and users, if in-cell formatting becomes a possibility. Perhaps in the future …
